options/options.go

```diff
@@ -1463,3 +1385,137 @@ func descriptorType(m proto.Message) string {
 		return fmt.Sprintf("%T", m)
 	}
 }
+
+func (interp *interpreter) messageLiteralValue(mc *internal.MessageContext, fieldNodes []*ast.MessageFieldNode, fmd protoreflect.MessageDescriptor) (interpretedFieldValue, error) {
+	fdm := dynamicpb.NewMessage(fmd)
+	origPath := mc.OptAggPath
+	defer func() {
+		mc.OptAggPath = origPath
+	}()
+	// NB: we don't want to leave this nil, even if the
+	// message is literal, because that indicates to
+	// caller that the result is not a message
+	flds := make([]*interpretedField, 0, len(fieldNodes))
+	for _, fieldNode := range fieldNodes {
+		if origPath == "" {
+			mc.OptAggPath = fieldNode.Name.Value()
+		} else {
+			mc.OptAggPath = origPath + "." + fieldNode.Name.Value()
+		}
+
+		// TODO: ensure that len(fieldNodes) == 1 (can't have multiple any fields)
+		if fieldNode.Name.IsAnyTypeReference() {
+			if fmd.FullName() == "google.protobuf.Any" {
+				// TODO: Support other URLs dynamically -- the caller of protoparse
+				// should be able to provide fldNode custom resolver that can resolve type
+				// URLs into message descriptors. The default resolver would be
+				// implemented as below, only accepting  "type.googleapis.com" and
+				// "type.googleprod.com" as hosts/prefixes and using the compiled
+				// file's transitive closure to find the named message.
+				urlPrefix := fieldNode.Name.URLPrefix.AsIdentifier()
+				msgName := fieldNode.Name.Name.AsIdentifier()
+				fullURL := fmt.Sprintf("%s/%s", urlPrefix, msgName)
+				if urlPrefix != "type.googleapis.com" && urlPrefix != "type.googleprod.com" {
+					return interpretedFieldValue{}, reporter.Errorf(interp.nodeInfo(fieldNode.Name.URLPrefix).Start(), "%vcould not resolve type reference %s", mc, fullURL)
+				}
+				anyFields, ok := fieldNode.Val.Value().([]*ast.MessageFieldNode)
+				if !ok {
+					return interpretedFieldValue{}, reporter.Errorf(interp.nodeInfo(fieldNode.Val).Start(), "%vtype references for google.protobuf.Any must have message literal value", mc)
+				}
+				anyMd := interp.file.ResolveMessageType(protoreflect.FullName(msgName))
+				if anyMd == nil {
+					return interpretedFieldValue{}, reporter.Errorf(interp.nodeInfo(fieldNode.Name.URLPrefix).Start(), "%vcould not resolve type reference %s", mc, fullURL)
+				}
+				// parse the message value
+				msgVal, err := interp.messageLiteralValue(mc, anyFields, anyMd)
+				if err != nil {
+					return interpretedFieldValue{}, err
+				}
+
+				// Any is defined with two fields:
+				//   string type_url = 1
+				//   bytes value = 2
+				typeURLDescriptor := fmd.Fields().ByNumber(1)
+				if typeURLDescriptor == nil {
```

Comment:
Just in case, to avoid a panic due to type mismatch
```suggestion
				if typeURLDescriptor == nil || typeURLDescriptor.Kind() != protoreflect.StringKind {
```